### PR TITLE
Townlist ToolUpdate/Down

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Complete

### CODE REVIEW

File: src/OpenLoco/GameCommands/GameCommands.h [hunk 1]
```diff
@@ -596,22 +596,56 @@ namespace OpenLoco::GameCommands
...
         {
         }
+        BuildingRemovalArgs(const BuildingPlacementArgs& place)
+            : pos(place.pos)
+            , type(place.type)
+        {
+        }
 
         Map::Pos3 pos;
+        uint8_t type;
```
Comment (duncanspumpkin) [inline src/OpenLoco/GameCommands/GameCommands.h:640]: investigate this change. Unsure may not be needed see right click

Comment (duncanspumpkin) [reply src/OpenLoco/GameCommands/GameCommands.h]: Confirmed should not be there.

**Review by AaronVanGeffen — COMMENTED** [review]
Buildings aren't built in the right colour. (Purple in preview, green on map)
Screenshot_20210722_155146

**Comment by duncanspumpkin** [comment]
Colour issues will be due to forgetting to pass in colour!

**Comment by duncanspumpkin** [comment]
That should fix the colour issues. Should probably make a changelog entry for fixing the dropdown.

File: src/OpenLoco/Windows/TownList.cpp [hunk 2]
```diff
@@ -970,26 +976,124 @@ namespace OpenLoco::Ui::Windows::TownList
...
+
+            auto z = surface->baseZ() * 4; //di
+            if (surface->slope())
+            {
+                z += 16;
+            }
+            args.pos = Map::Pos3(pos->x, pos->y, z);
+            args.type = townListWnd->row_hover;  //dl
+            args.variation = _buildingVariation; //dh
+            if (isEditorMode())
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/TownList.cpp:1033]: Can we do this in sandbox mode too? Worth a changelog entry if we do.

**Review by AaronVanGeffen — APPROVED** [review]
Working properly. Left one suggestion; up to you if you'd like to tackle that in this PR.

**Pull request merged**